### PR TITLE
Small updates to F# recursion article

### PR DESCRIPTION
**Pull request by cartermp**
Clarifies a few more things and adds more examples.

### CODE REVIEW

File: docs/fsharp/language-reference/functions/recursive-functions-the-rec-keyword.md [hunk 1]
```diff
@@ -12,28 +12,45 @@ The `rec` keyword is used together with the `let` keyword to define a recursive
...
+    function2-body
 ...
 ```
 
 ## Remarks
 
-Recursive functions, functions that call themselves, are identified explicitly in the F# language. This makes the identifer that is being defined available in the scope of the function.
+Recursive functions - functions that call themselves - are identified explicitly in the F# language with the `rec` keyword. The `rec` keyword makes the name of the `let` binding available in its body.
+
+The following example shows a recursive function that computes the *n*<sup>th</sup> Fibonacci number using the mathematical definition.
```
Comment (Youssef1313) [inline docs/fsharp/language-reference/functions/recursive-functions-the-rec-keyword.md:31]: Fixes markdownlint error.

```suggestion
```

File: docs/fsharp/language-reference/functions/recursive-functions-the-rec-keyword.md [hunk 2]
```diff
@@ -70,6 +87,14 @@ The following example shows two mutually recursive functions.
 
 [!code-fsharp[Main](~/samples/snippets/fsharp/lang-ref-1/snippet4002.fs)]
 
+## Recursive values
+
+You can also define a `let`-bound value to be recursive. This is sometimes done for logging. With F# 5 and the `nameof` function, you can do this:
```
Comment (Youssef1313) [inline docs/fsharp/language-reference/functions/recursive-functions-the-rec-keyword.md:92]: In C#, `nameof` is an operator not a function. I don't know about F#, but just wanted to note.

Comment (cartermp) [reply docs/fsharp/language-reference/functions/recursive-functions-the-rec-keyword.md:91]: The distinction between operators and functions in F# isn't quite so clear. But it's defined as a function.

**Review by BillWagner — APPROVED** [review]
This LGTM. You can :shipit: when ready.

**Pull request merged**